give-addon-boilerplate.php

```diff
@@ -13,31 +18,38 @@
 
 // Add-on name
 define( 'ADDON_CONSTANT_NAME', 'ADDON_NAME' );
+
 // Versions
-define( 'ADDON_CONSTANT_VERSION',          '1.0.0' );
+define( 'ADDON_CONSTANT_VERSION', '1.0.0' );
 define( 'ADDON_CONSTANT_MIN_GIVE_VERSION', '2.8.0' );
+
 // Add-on paths
-define( 'ADDON_CONSTANT_FILE',     __FILE__ );
-define( 'ADDON_CONSTANT_DIR',      plugin_dir_path( ADDON_CONSTANT_FILE ) );
-define( 'ADDON_CONSTANT_URL',      plugin_dir_url( ADDON_CONSTANT_FILE ) );
+define( 'ADDON_CONSTANT_FILE', __FILE__ );
+define( 'ADDON_CONSTANT_DIR', plugin_dir_path( ADDON_CONSTANT_FILE ) );
+define( 'ADDON_CONSTANT_URL', plugin_dir_url( ADDON_CONSTANT_FILE ) );
 define( 'ADDON_CONSTANT_BASENAME', plugin_basename( ADDON_CONSTANT_FILE ) );
 
-require ADDON_CONSTANT_DIR . 'vendor/autoload.php';
+require './vendor/autoload.php';
 
 // Activate add-on hook.
-register_activation_hook( ADDON_CONSTANT_FILE, [ Addon\Activation::class, 'activateAddon' ] );
+register_activation_hook( ADDON_CONSTANT_FILE, [ Activation::class, 'activateAddon' ] );
+
 // Deactivate add-on hook.
-register_deactivation_hook(	ADDON_CONSTANT_FILE, [ Addon\Activation::class, 'deactivateAddon' ] );
+register_deactivation_hook( ADDON_CONSTANT_FILE, [ Activation::class, 'deactivateAddon' ] );
+
 // Uninstall add-on hook.
-register_uninstall_hook( ADDON_CONSTANT_FILE, [ Addon\Activation::class, 'uninstallAddon' ] );
+register_uninstall_hook( ADDON_CONSTANT_FILE, [ Activation::class, 'uninstallAddon' ] );
 
 // Register the add-on service provider with the GiveWP core.
-add_action( 'before_give_init', function() {
-	// Check Give min required version.
-	if ( Addon\Helpers\Environment::giveMinRequiredVersionCheck() ) {
-		give()->registerServiceProvider( Addon\AddonServiceProvider::class );
+add_action(
```

Comment:
I (personally) like having this as a one-off `add_action` instead of wrapping it in a larger class (which is really just organization and not object oriented).

Reply:
@kjohnson Just to be clear, are you saying pulling the code out of the `Environment::giveMinRequiredVersionCheck` and just put it inline here?

Reply:
Ah, no. I was just commenting that I liked your use of add_action here, as opposed to creating an organizational class. Procedural works well here.